tests/e2e/tidbcluster/across-kubernetes.go

```diff
@@ -240,6 +241,329 @@ var _ = ginkgo.Describe("[Across Kubernetes]", func() {
+
+			ginkgo.By("Installing tidb server and client certificate")
+			err = InstallXK8sTiDBCertificates(ns1, tcName1, clusterDomain)
+			framework.ExpectNoError(err, "failed to install tidb server and client certificate for cluster: %q", tcName1)
+			err = InstallXK8sTiDBCertificates(ns2, tcName2, clusterDomain)
+			framework.ExpectNoError(err, "failed to install tidb server and client certificate for cluster: %q", tcName2)
+			err = InstallXK8sTiDBCertificates(ns3, tcName3, clusterDomain)
+			framework.ExpectNoError(err, "failed to install tidb server and client certificate for cluster: %q", tcName3)
+
+			ginkgo.By("Installing tidb components certificates")
+			err = InstallXK8sTiDBComponentsCertificates(ns1, tcName1, clusterDomain)
+			framework.ExpectNoError(err, "failed to install tidb components certificates for cluster: %q", tcName1)
+			err = InstallXK8sTiDBComponentsCertificates(ns2, tcName2, clusterDomain)
+			framework.ExpectNoError(err, "failed to install tidb components certificates for cluster: %q", tcName2)
+			err = InstallXK8sTiDBComponentsCertificates(ns3, tcName3, clusterDomain)
+			framework.ExpectNoError(err, "failed to install tidb components certificates for cluster: %q", tcName3)
+
+			ginkgo.By("Creating x-k8s tidb clusters with TLS enabled")
+			tc1.Spec.TiDB.TLSClient = &v1alpha1.TiDBTLSClient{Enabled: true}
+			tc1.Spec.TLSCluster = &v1alpha1.TLSCluster{Enabled: true}
+			utiltc.MustCreateTCWithComponentsReady(genericCli, oa, tc1, 10*time.Minute, 30*time.Second)
+
+			tc2.Spec.TiDB.TLSClient = &v1alpha1.TiDBTLSClient{Enabled: true}
+			tc2.Spec.TLSCluster = &v1alpha1.TLSCluster{Enabled: true}
+			utiltc.MustCreateTCWithComponentsReady(genericCli, oa, tc2, 10*time.Minute, 30*time.Second)
+
+			tc3.Spec.TiDB.TLSClient = &v1alpha1.TiDBTLSClient{Enabled: true}
+			tc3.Spec.TLSCluster = &v1alpha1.TLSCluster{Enabled: true}
+			utiltc.MustCreateTCWithComponentsReady(genericCli, oa, tc3, 10*time.Minute, 30*time.Second)
+
+			ginkgo.By("Fail TiKV in cluster-1 by setting a wrong image")
+			local, err := cli.PingcapV1alpha1().TidbClusters(tc1.Namespace).Get(context.TODO(), tc1.Name, metav1.GetOptions{})
+			framework.ExpectNoError(err, "getting tidbcluster %s/%s", tc1.Namespace, tc1.Name)
+			local.Spec.TiKV.BaseImage = inexistentBaseImage
+			err = genericCli.Update(context.TODO(), local)
+			framework.ExpectNoError(err, "updating tikv with an inexistent image %q for %q", tc1.Spec.TiKV.BaseImage, tcName1)
+			// force operator to trigger an upgrade.
+			err = c.AppsV1().StatefulSets(ns1).Delete(context.TODO(), fmt.Sprintf("%s-tikv", tcName1), metav1.DeleteOptions{})
+			framework.ExpectNoError(err, "deleting sts of tikv for %q", tcName1)
+
+			ginkgo.By("Waiting for tikv pod to be unavailable")
+			err = utiltc.WaitForTidbClusterCondition(cli, tc1.Namespace, tc1.Name, time.Minute*5, func(tc *v1alpha1.TidbCluster) (bool, error) {
+				for _, store := range tc.Status.TiKV.Stores {
```

Comment:
We need to check for all TiKVs here, right?

Reply:
For this case, it's 1 replica only. I'll change this by checking all stores' state.